### PR TITLE
Feature/3d and 2d

### PR DESCRIPTION
**Pull request by yunse-lee-EPITECH**
fix: #156

### CODE REVIEW

File: libs/engine/includes/Registry.hpp [hunk 1]
```diff
@@ -18,12 +18,27 @@
 #include <functional>
 #include <list>
 #include <memory>
+#include <optional>
+#include <queue>
```
Comment (Snowy1803) [inline libs/engine/includes/Registry.hpp:22]: clang format ?

File: libs/engine/includes/Registry.hpp [hunk 2]
```diff
@@ -18,12 +18,27 @@
...
 #include <typeindex>
 #include <unordered_map>
 #include <vector>
-#include <queue>
 
 namespace engine {
+struct TextureWrapper {
+    std::string path_2d;
+    std::optional<std::string> path_3d;
```
Comment (Snowy1803) [inline libs/engine/includes/Registry.hpp:30]: Camel case

File: libs/rendering/sources/Systems/Texture.cpp [hunk 3]
```diff
@@ -7,26 +7,51 @@
 
 #include "Systems/Texture.hpp"
 #include <algorithm>
+#include <cmath>
+#include <iostream>
```
Comment (Snowy1803) [inline libs/rendering/sources/Systems/Texture.cpp:11]: Why

File: libs/rtype/sources/RTypeGame.cpp [hunk 4]
```diff
@@ -30,6 +30,8 @@
 #include "NetworkClientSystem.hpp"
 #include "Rendering.hpp"
 
+#include <iostream>
```
Comment (Snowy1803) [inline libs/rtype/sources/RTypeGame.cpp:33]: Why

File: libs/rtype/sources/RTypeGame.cpp [hunk 5]
```diff
@@ -289,7 +307,7 @@ void RTypeGame::initScene(engine::Registry &reg)
         reg.get_components<Component::Drawable>().insert_at(
             player,
             Component::Drawable(
-                SHIP_I, SHIP_W, SHIP_H, 3, 17 * (client.getPlayerNumber() - 1)
+                11, SHIP_W, SHIP_H, 3, 17 * (client.getPlayerNumber() - 1)
```
Comment (Snowy1803) [inline libs/rtype/sources/RTypeGame.cpp:310]: Why

Comment (yunse-lee-EPITECH) [reply libs/rtype/sources/RTypeGame.cpp:310]: comment ça why

File: libs/engine/includes/Registry.hpp (hunk 1 above)
Comment (yunse-lee-EPITECH) [reply libs/engine/includes/Registry.hpp:22]: ??? c bien en ordre?

**Review by jeremyelalouf — APPROVED** [review]

**Pull request merged**